pkg/operator/ceph/object/objectstore.go

```diff
@@ -1000,3 +1009,46 @@ func errorOrIsNotFound(err error, msg string, args ...string) error {
 	}
 	return errors.Wrapf(err, msg, args)
 }
+
+// ShouldUpdateZoneEndpointList checks whether zone endpoint list need to be updated or not
+func ShouldUpdateZoneEndpointList(zones []zoneType, desiredEndpointList []string, zoneName string) (bool, error) {
+	if zoneName == "" {
+		return false, errors.Errorf("zone name can't be empty")
+	}
+
+	zoneExists, endpoints := findZoneEndpoints(zoneName, zones)
+	if !zoneExists ||
+		(len(desiredEndpointList) == 0 && len(endpoints) == 0) {
+		return false, nil
+	}
+
+	return !listsAreEqual(desiredEndpointList, endpoints), nil
+}
+
+func findZoneEndpoints(targetZone string, zones []zoneType) (bool, []string) {
+	for _, z := range zones {
+		if z.Name == targetZone {
+			return true, z.Endpoints
+		}
+	}
+	return false, []string{}
+}
+
+func listsAreEqual(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	as := make([]string, len(a))
+	bs := make([]string, len(b))
+	copy(as, a)
+	copy(bs, b)
+	sort.Strings(as)
+	sort.Strings(bs)
+	for i := 0; i < len(a); i++ {
+		if as[i] != bs[i] {
+			return false
```

Comment:
if the lists has duplicates you will report as "false" even iof the lists are the same.
probably better to convert to a "set" (a golang map without values), and compare the 2 sets.

Reply:
The idea of duplicates is a good one. Thanks Yuval. IMO, I think this function is fine if we give it de-duplicated inputs. And we likely want to check for duplicates beforehand. 

Instead, I think we should check for duplicates in the list the user gives and flag that as an error to the user. 

Similarly, if there are duplicates in the endpoints list from the RGW, we should probably just remove the duplicates.

Reply:
If the endpoint list is populated by rook(based on rgw service name), then there is no chance of having duplicate entries. Because the endpoint is added only if the entry is missing in the existing list. The user inputs duplicate entries in `customEndpoints` we can error out. But honestly IMO even if we store duplicates entries to endpoint list in zone/zonegroup, AFAIK I can't see any functional impact due to that for multisite to work. @alimaredia or @cbodley ??

Reply:
> But honestly IMO even if we store duplicates entries to endpoint list in zone/zonegroup, AFAIK I can't see any functional impact due to that for multisite to work. @alimaredia or @cbodley ??

you're right that duplicate endpoints won't break rgw's replication, but it's also not optimal. we do a round robin over these endpoints, so duplicates would mean the work doesn't spread evenly